tests/test_averaging.py

```diff
@@ -150,15 +150,15 @@ def test_allgather():
 @pytest.mark.asyncio
 async def test_allreduce_protocol():
```

Comment:
I decided not to make a separate test since there will be a lot of copy paste. Is it ok or there are some arguments for making a separate one?

Reply:
I am okay with whichever version you prefer here.
@mryab , do you have any preferences?

Reply:
One test is fine, I guess (as long as we check that the colab peer gets the average value)